src/App.js

```diff
@@ -16,44 +16,59 @@ import { Component } from "react";
 import classPage from "./components/classPage/classPage";
 import ScrollToTop from "./components/ScrollToTop";
 
-const App = () => {
-  const l = api.book.read();
-  console.log(l);
-  const routes = [
-    { path: "/category/:name", as: Category },
-    {
-      path: "/book/:id",
-      as: (props) => {
-        const id = props.location.state.id;
-        return Book(MOCK_DATA[id - 1]);
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      books: {},
+      categories: {},
+    };
+  }
+
+  componentDidMount() {
```

Comment:
the api.book.<method> library will automatically populate the redux store with the requested items. that way, if another component needs access to the books, we won't duplicate requests.

```js
function App(props){
  !empty(props.categories) || api.category.get()
...
export default connect(
  (state, props) => ({ categories: state.category, books:state.book, ...props }),
  {}
)(App);```